python/lsst/ap/association/transformDiaSourceCatalog.py

```diff
@@ -204,34 +204,29 @@ def run(self,
             "Transforming/standardizing the DiaSource table ccdVisitId: %i",
             ccdVisitId)
 
-        diaSourceDf = diaSourceCat.asAstropy().to_pandas()
-
         def getSignificance():
             """Return the significance value of the first peak in each source
-            footprint."""
-            size = len(diaSourceDf)
-            result = np.full(size, np.nan)
-            for i in range(size):
-                record = diaSourceCat[i]
-                if self.config.doRemoveSkySources and record["sky_source"]:
-                    continue
+            footprint, or NaN for peaks without a significance field.
+            """
+            result = np.full(len(diaSourceCat), np.nan)
+            for i, record in enumerate(diaSourceCat):
                 peaks = record.getFootprint().peaks
                 if "significance" in peaks.schema:
                     result[i] = peaks[0]["significance"]
             return result
 
+        diaSourceDf = diaSourceCat.asAstropy().to_pandas()
         diaSourceDf["snr"] = getSignificance()
-
-        if self.config.doRemoveSkySources:
-            diaSourceDf = diaSourceDf[~diaSourceDf["sky_source"]]
-
         diaSourceDf["bboxSize"] = self.computeBBoxSizes(diaSourceCat)
         diaSourceDf["ccdVisitId"] = ccdVisitId
         diaSourceDf["filterName"] = band
         diaSourceDf["midPointTai"] = diffIm.getInfo().getVisitInfo().getDate().get(system=DateTime.MJD)
         diaSourceDf["diaObjectId"] = 0
         diaSourceDf["ssObjectId"] = 0
 
+        if self.config.doRemoveSkySources:
+            diaSourceDf = diaSourceDf[~diaSourceDf["sky_source"]]
+
```

Comment:
Why change the  order here? It seems a bit silly to do the column updates before trimming out the rows.

Reply:
I agree that updating all the columns and then removing some rows isn't ideal, but it made the code (especially `getSignificance` and `computeBBoxSizes`) a lot clearer, I think.

Having the sky source removal happen in the middle of the source assignment (`diaSourceDf["snr"] = getSignificance()` needs to happen before sky sources are removed, and the previous version of `computeBBoxSizes()` had to happen after) made it more fragile: it wasn't immediately obvious where to insert a new computed field. Doing all the sky source removal at once at the end was much simpler than having extra logic inside those two methods to determine whether to process a sky source or not. I don't think has any significant effect on the runtime.

Reply:
It turns out that you can do boolean slicing on afw catalogs! I've pulled out `getSignificance` to a free function and I do the filtering first now. I think that is an improvement.